app/controllers/sessions_controller.rb

```diff
@@ -10,7 +10,11 @@ class Stringer < Sinatra::Base
     if user
       session[:user_id] = user.id
 
-      redirect to("/")
+      if session[:redirect_to].present?
```

Comment:
Small opportunity to get rid of the conditional here, by specifying a default `redirect_to` like so:

```ruby
redirect_uri = session.delete(:redirect_to) || '/'
redirect to(redirect_uri)
```

Alternatively you could use the block syntax to specify a default, like so:

```ruby
session.delete(:redirect_to) { '/' }
```

I personally prefer the first, without block.